Controllers/StockController.cs

```diff
@@ -0,0 +1,282 @@
+                {
+                    new Int32Converter(),
+                    new LongConverter(),
+                    new DecimalConverter(),
+                    new DateTimeOffsetConverter()
+                }
+            };
+        }
+
+        [HttpGet]
+        public ActionResult Symbols(string exchange)
+        {
+            try
+            {
+                var parameters = $"https://finnhub.io/api/v1/stock/symbol?exchange={exchange.ToUpperInvariant()}";
+
+                HttpResponseMessage response = finnhubClient.GetAsync(parameters).Result;
+                if (!response.IsSuccessStatusCode)
+                {
+                    throw new Exception($"Status code: {response.StatusCode}");
+                }
+
+                List<StockSymbol> stockSymbol =
+                    response.Content.ReadFromJsonAsync<List<StockSymbol>>(serializerOptions).Result;
+
+                return Ok(stockSymbol);
+            }
+            catch (Exception e)
+            {
+                return BadRequest($"Request failed. {e.Message}");
+            }
+        }
+
+        [HttpGet("{symbol}/profile")]
+        public ActionResult Profile(string symbol)
+        {
+            try
+            {
+                var parameters = $"https://finnhub.io/api/v1/stock/profile2?symbol={symbol.ToUpperInvariant()}";
+
+                HttpResponseMessage response = finnhubClient.GetAsync(parameters).Result;
+                if (!response.IsSuccessStatusCode)
+                {
+                    throw new Exception($"Status code: {response.StatusCode}");
+                }
+
+                CompanyProfile2 profile = response.Content.ReadFromJsonAsync<CompanyProfile2>(serializerOptions).Result;
+
+                if (profile?.Ticker is null)
+                {
+                    throw new Exception("Symbol is invalid or profile does not exist.");
+                }
+
+                profile.Exchange = GetExchangeAcronym(profile.Exchange);
+
+                return Ok(profile);
+            }
+            catch (Exception e)
+            {
+                return BadRequest($"Request failed. {e.Message}");
+            }
+        }
+
+        [HttpGet("{symbol}/sentiment")]
+        public ActionResult Sentiment(string symbol)
+        {
+            try
+            {
+                var parameters = $"https://finnhub.io/api/v1/news-sentiment?symbol={symbol.ToUpperInvariant()}";
+
+                HttpResponseMessage response = finnhubClient.GetAsync(parameters).Result;
+                if (!response.IsSuccessStatusCode)
+                {
+                    throw new Exception($"Status code: {response.StatusCode}");
+                }
+
+                NewsSentiment sentiment = response.Content.ReadFromJsonAsync<NewsSentiment>(serializerOptions).Result;
+
+                if (sentiment?.Symbol is null)
+                {
+                    throw new Exception("Symbol is invalid or news sentiment does not exist.");
```

Comment:
Codacy found an issue: ['System.Exception' should not be thrown by user code.](https://app.codacy.com/gh/InvestingOak/InvestingOak/pullRequest?prid=6380270)